compiler/luci/pass/src/FoldDequantizePass.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldDequantizePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <loco/Service/TypeInference.h>
+
+namespace
+{
+
+bool is_hybrid_kernel_supported(loco::Node *node)
+{
+  if (dynamic_cast<luci::CircleFullyConnected *>(node) != nullptr)
+    return true;
+
+  return false;
+}
+
+bool is_foldable_const(luci::CircleConst *node)
+{
+  if (node->dtype() == loco::DataType::S8)
+    return true;
+  if (node->dtype() == loco::DataType::U8)
+    return true;
+
+  return false;
+}
+
+luci::CircleConst *dequantized_const_node(luci::CircleConst *const_node)
+{
+  auto g = const_node->graph();
+  auto new_const_node = g->nodes()->create<luci::CircleConst>();
+
+  new_const_node->dtype(loco::DataType::FLOAT32);
+  new_const_node->rank(const_node->rank());
+  uint32_t dim_size = 1;
+  for (uint32_t i = 0; i < new_const_node->rank(); ++i)
+  {
+    new_const_node->dim(i) = const_node->dim(i);
+    dim_size *= const_node->dim(i).value();
+  }
+  new_const_node->size<loco::DataType::FLOAT32>(dim_size);
+  new_const_node->shape_status(luci::ShapeStatus::VALID);
+
+  const int32_t q_dim = const_node->quantparam()->quantized_dimension;
```

Comment:
Please add `assert(const_node->quantparam() != nullptr)` at top of this method

Reply:
Hmm... I am okay to add `assert` but can you explain the difference with https://github.com/Samsung/ONE/pull/4434#discussion_r495626783?

FYI, `const_node->quantparam()` is being checked at https://github.com/Samsung/ONE/pull/4404/files#diff-378da9ae08b1d64c34d114cfebd0da6cR169 :)

Reply:
> I am okay to add assert but can you explain the difference with #4434 (comment)?

For this method only, there should be some condition for `const_node->quantparam()` is not null.
You can also return if this condition is not met.

For `difference with #4434 (comment)?`: 
1) if the assert in #4434 can happen in any model then Pass may assert in debug that may break the execution; it would be better to just skip for that node. For release version, the result could be unpredeictable.
2) if this doesn't happen in real models but you want to be sure for debugging; it is like debug condition, then it is OK. But the code is like `if {condition) ... else assert()` which doesn't seem to be for this case. For the logic, I think it would be better with something like `if (condition) ... else { LOG(...); continue }` or `assert(condition);` with some comment `This conditon must always be true`

Reply:
Thanks for explanation, I will add `assert` here :)